src/renderer/shaders/geometry/polygon/polygonVertexShader.glsl

```diff
@@ -20,17 +20,23 @@ float decodeWidth(vec2 enc) {
 }
 
 void main(void) {
+    // 64 is computed based on RTT_WIDTH and the depth buffer precision
+    // 64 = 2^(BUFFER_BITS)/RTT_WIDTH = 2^16/1024 = 64
+    float z = mod(featureID.y, 1./64.)*63. + featureID.x / (64.);
+
     vec4 c;
     if (normal == vec2(0.)){
+        z = 2.*z - 1.;
         c = texture2D(colorTex, featureID);
     }else{
+        z = 2.*z - 1. + 1./128.;
```

Comment:
So... this uses a separate set of depth values for the stroke than for the filling, right?

Reply:
Yes, we need different z inside the same feature for fill and stroke. Since we are moving z (0, 1) to (-1, 1) I think we can cover the gaps with the z index for stroke and I think that's what this code does, but please take a closer look to be sure about it :wink:

Reply:
I think this line can overflow the valid `z` range of [-1,1]. I think we could leave the `z = 2.*z - 1.;` after both branches, and in the `else` branch have: `z = mod(z-0.5, 1.);`